plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/EditResourceLimitsPage.java

```diff
@@ -0,0 +1,119 @@
+        createResourceWidgets(dbc, group, OpenShiftUIMessages.RequestLabel, EditResourceLimitsPageModel.REQUESTS_MEMORY, MEMORY_SUFFIXES, MEMORY_SUFFIXES_LABELS);
+        createResourceWidgets(dbc, group, OpenShiftUIMessages.LimitLabel, EditResourceLimitsPageModel.LIMITS_MEMORY, MEMORY_SUFFIXES, MEMORY_SUFFIXES_LABELS);
+        
+        group = new Group(dialogArea, SWT.NONE);
+        group.setText(OpenShiftUIMessages.CPULabel);
+        GridDataFactory.fillDefaults().span(3, 1).grab(true, false).align(SWT.FILL, SWT.FILL).applyTo(group);
+        GridLayoutFactory.fillDefaults().numColumns(3).margins(10, 20).spacing(10, 4).applyTo(group);
+        createResourceWidgets(dbc, group, OpenShiftUIMessages.RequestLabel, EditResourceLimitsPageModel.REQUESTS_CPU, CPU_SUFFIXES, CPU_SUFFIXES_LABELS);
+        createResourceWidgets(dbc, group, OpenShiftUIMessages.LimitLabel, EditResourceLimitsPageModel.LIMITS_CPU, CPU_SUFFIXES, CPU_SUFFIXES_LABELS);
+    }
+    
+    private void createResourceWidgets(DataBindingContext dbc, Group group, String label, String property, String[] suffixes, String[] labels) {
+        // label
+        Label labelComp = new Label(group, SWT.NONE);
+        labelComp.setText(label);
+        GridDataFactory.fillDefaults().align(SWT.FILL, SWT.CENTER).applyTo(labelComp);
+
+        // scale spinner
+        Text text = new Text(group, SWT.BORDER);
+        GridDataFactory.fillDefaults().align(SWT.FILL, SWT.FILL).grab(true, false).applyTo(text);
+        ComboViewer combo = new ComboViewer(group);
+        combo.setContentProvider(ArrayContentProvider.getInstance());
+        combo.setInput(suffixes);
+        combo.setLabelProvider(new LabelProvider() {
+
+            @Override
+            public String getText(Object element) {
+                for(int i=0; i < suffixes.length;++i) {
+                    if (element.equals(suffixes[i])) {
+                        element = labels[i];
+                        break;
+                    }
+                }
+                return (String) element;
+            }
+            
+        });
+
+        IObservableValue textObservable = WidgetProperties.text(SWT.Modify).observe(text);
+        IObservableValue comboObservable = ViewerProperties.singleSelection().observe(combo);
+        IObservableValue master = BeanProperties.value(EditResourceLimitsPageModel.SELECTED_CONTAINER).observe(model);
+        ValueBindingBuilder.bind(textObservable)
+                .validatingAfterGet(new NumericValidator("integer", Integer::parseInt))
+                .converting(new AggregatingConverter(comboObservable, true))
+                .to(PojoProperties.value(property).observeDetail(master))
+                .converting(new KeywordConverter(suffixes, true))
+                .in(dbc);
+        ValueBindingBuilder.bind(comboObservable)
+        .converting(new AggregatingConverter(textObservable, false))
+        .to(PojoProperties.value(property).observeDetail(master))
```

Comment:
What's the reason to use PojoProperties and not BeanProperties?

Reply:
Because the bound object is an IContainer that does not support listeners